nucleus/core/kernel/src/main/java/com/sun/enterprise/v3/services/impl/GlassfishNetworkListener.java

```diff
@@ -374,8 +374,9 @@ public GlassfishHttpCodecFilter(
             serverVersion = serverInfo == null ? Version.getProductId() : serverInfo;
 
             if (isXPoweredByEnabled) {
```

Comment:
Why not just disable the whole `X-Powered-By` header if `product.name` is empty? I'd expect this is what users would expect if they set it to an empty value.

```suggestion
            if (isXPoweredByEnabled || (serverInfo != null && serverInfo.isEmpty()) {
```

Reply:
Thank you for your suggestion.  
I described the past GlassFish behavior as expected one, but your idea would be better.  
I've corrected it.